extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts

```diff
@@ -569,7 +569,8 @@ export class PublishDatabaseDialog {
 
 				const data = this.convertSqlCmdVarsToTableFormat(this.getSqlCmdVariablesForPublish());
 				await (<azdata.DeclarativeTableComponent>this.sqlCmdVariablesTable).updateProperties({
-					data: data
+					dataValues: data,
+					data: [] // data is deprecated, but the table gets updated incorrectly if this isn't set to an empty array
```

Comment:
I'd prefer to go to the component and make sure data has a proper default that it needs instead of forcing the extensions to provide an empty array.

Reply:
updated to fix this in the declarativeTableComponent so the empty array doesn't need to be passed in